### DISCUSSION TITLE
Issues quering Entraid groups

### DISCUSSION THREAD

**Discussion by CoreyS222** [Set-OutlookSignatures/Set-OutlookSignatures#121] [Q&A] [answered]
Hi Everyone. 
we are lookign to implement solution this as a replacement for an out of date piece of software. 
We are running in a Hybrid AD environment with Windows 10 and 11 as cloud only or domain clients that are sync'd to AAD and full outlook client installs. 
We are slowly moving to Cloud only device so i need to get this working for using graph only. 
Running the script with the Graph only tag pulls back all the correct user details but when processing the INI file the output shows Unknown Tag when using the EntraID <groupdisplayname> or AzureAD <groupdisplayname> 
I have tried Cloud native and Hybrid groups but i get the same result. either all the signatures in the Ini file are assigned to the user account or none of them are. 

PS command 
.\set-outlooksignatures.ps1  -SignatureTemplatePath ".\templates" -UseHtmTemplates $true -SignatureIniPath ".\Templates\_Signatures.ini" -SetCurrentUserOOFMessage $False -SetCurrentUserOutlookWebSignature $false -Graphonly $true -verbose
Result from  Powershell 

'Sig-Rus.htm(signature ini index #1)
      Common template (no group, email address or replacement variable allow tags specified)
      Default signature for new emails
        defaultNew
      Default signature for replies and forwards
        defaultReplyFwd
      Unknown tags
        EntraID Sig-Rus

'Rus-Sig.htm' (Signature ini index #1) @2024-08-01T12:17:17+12:00@
      Check permissions
        Allows
          No group match for current mailbox, checking current user specific allows
          Group: Mailbox and current user are not member of any allowed group
        Do not use template as there is no allow or at least one deny

- ---ini file --- -
# Internal signatures
## Internal signature, common
[Sig-Rus.htm]
EntraID Sig-Rus
defaultNew
defaultReplyFwd

Hopely someone can point me in the right direction

**Comment by GruberMarkus** [comment]
"Sig-Rus" in "EntraID Sig-Rus" is not an email address, so - according to the README file - it must either be the display name or the mailNickname of the group. Which one is it?


For further analysis, please run Set-OutlookSignatures with the "-verbose" parameter and provide the full unmodified text output.

**Comment by GruberMarkus** [comment]
Good morning!

1. The log file has been modified. I assume you only changed information that is not required to answer your question.
2. You are not using the latest release of Set-OutlookSignatures.
3. You are not only querying Graph. As `-GraphOnly true` is not set, Set-OutlookSignatures tries to connect to your on-prem AD first. In the log file, this does not work (probably because there is no connection to a domain controller), therefore the software switches to Graph.
4. Sig-Rus.htm does not exist in the signature template path, so it can never be converted to a signature. The remark in the log is:
   ```
   Get all signature template files and categorize them @2024-08-05T08:50:34+12:00@
     Compare signature ini entries and file system
       'Sig-Rus.htm' (signature ini index #1) found in ini but not in signature template path.
   ```
5. One line later in the log, another discrepancy between SignatureTemplatePath and SignatureIniFile is logged:
   ```
   'Russley Road.htm' found in signature template path but not in ini.
   ```
7. The following exclusion does not work as the Object ID does not match one of the formats defined in the README file: -CURRENTUSER:EntraID <email-address-of-group@example.com>, -CURRENTUSER:EntraID <GroupMailNickname>, -CURRENTUSER:EntraID <GroupDisplayName>. 'Russley Road2.htm' is applied to all mailboxes because it is a common template with a non-working exclusion.
   ```
   [Russley Road2.htm]
   -currentuser:EntraID fbd44348-f244-42af-8ed7-b54dab06da2e
   ```

I recommend correcting your INI file and making sure the referenced template filesexist in SignatureTemplatePath.

**Comment by GruberMarkus** [comment]
@CoreyS222: Does the previous comment answer your questions and solve the issues?

**Reply by CoreyS222** [reply]
We have corrected the INI File, and this has fixed some of our issues.
However we have discovered that most of the groups we are planning in using have spaces in the display names. This seems to break the script. 
We are looking at removing the spaces in these groups.

**Comment by GruberMarkus** [comment, marked as answer]
You are correct: Spaces in group names are not handled correctly because of a regression. Thank you for reporting!

You can fix this yourself until the next version is published: In `Set-OutlookSignatures.ps1`, search for
```
"(displayName eq '$($local:NTName.Split('\')[1])'))"
```
and change it to 
```
"(displayName eq '$($local:NTName.Split('\')[1])')"
```

Do I understand correctly that all issues are fixed now?

**Comment by GruberMarkus** [comment]
Closing because @CoreyS222 did not respond for five days.
